optuna/integration/sklearn.py

```diff
@@ -55,30 +47,53 @@
     from typing import Optional  # NOQA
     from typing import Union  # NOQA
 
+    ArrayLikeType = Union[List, np.ndarray, pd.Series]
     OneDimArrayLikeType = Union[List[float], np.ndarray, pd.Series]
     TwoDimArrayLikeType = \
         Union[List[List[float]], np.ndarray, pd.DataFrame, spmatrix]
 
 _logger = logging.get_logger(__name__)
 
 
+def _is_arraylike(x):
+    # type: (Any) -> bool
+
+    return (
+        hasattr(x, '__len__') or
+        hasattr(x, 'shape') or
+        hasattr(x, '__array__')
+    )
+
+
+def _num_samples(x):
```

Comment:
Original implementation is [here](https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/utils/validation.py#L138-L163).
How much should I take care of the original implementation? (is this too simple?)

Reply:
It basically seems good to me to keep the code simple.
This implementation removes the handling of exceptional cases, so we may have bug reports about them.
So, please add a link to the original implementation (including the commit id) as a comment.

Also, I'm not familiar with Dask dataframe, but I think we need the following check for Dask users.
https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/utils/validation.py#L155-L158

Reply:
Thank you for the suggestion. :pray:
I added the specific check for a dask dataframe in f8f21a5.

Reply:
Thank you for the suggestion. :pray:
I added the specific check for a dask dataframe in f8f21a5.